File: Problems/No.1893_Check_if_All_the_Integers_in_a_Range_Are_Covered/check_if_all_the_integers_in_a_range_are_covered.cpp
class Solution {
public:
    bool isCovered(vector<vector<int>>& ranges, int left, int right) {
        
        // parameter
        std::vector<bool> check_range(right-left+1, false);

        // start computing
        for (auto &range: ranges) {

            int rangeL = range[0];
            int rangeR = range[1];
            for (int i=std::max(rangeL, left); i<=std::min(rangeR, right); i++)
                check_range[i-left] = true;

        }

        for (auto check_int: check_range) if (!check_int) return false;
        return true;

    }
};

File: Problems/No.1893_Check_if_All_the_Integers_in_a_Range_Are_Covered/check_if_all_the_integers_in_a_range_are_covered.py
class Solution(object):
    def isCovered(self, ranges, left, right):
        """
        :type ranges: List[List[int]]
        :type left: int
        :type right: int
        :rtype: bool
        """

        # parameter
        check_range = [False] * (right-left+1)

        # start computing
        for [rangeL, rangeR] in ranges:
            for i in range(max(rangeL, left), min(rangeR, right)+1):
                check_range[i-left] = True
        
        return True if False not in check_range else False
